Mathlib/Topology/Algebra/Order/Floor.lean

```diff
@@ -27,8 +26,36 @@ This file proves statements about limits and continuity of functions involving `
 
 open Filter Function Int Set Topology
 
+namespace FloorSemiring
+
+open scoped Nat
+
+variable {K : Type*} [LinearOrderedField K] [FloorSemiring K] [TopologicalSpace K] [OrderTopology K]
+
+theorem tendsto_mul_pow_div_factorial_sub (a c : K) (d : ℕ) :
+    Tendsto (fun n ↦ a * c ^ n / (n - d)!) atTop (𝓝 0) := by
+  rw [tendsto_order]
+  constructor
+  all_goals
+    intro ε hε
+    filter_upwards [eventually_mul_pow_lt_factorial_sub (a * ε⁻¹) c d] with n h
+    rw [mul_right_comm, ← div_eq_mul_inv] at h
+  · rw [div_lt_iff_of_neg hε] at h
+    rwa [lt_div_iff' (Nat.cast_pos.mpr (Nat.factorial_pos _))]
+  · rw [div_lt_iff hε] at h
+    rwa [div_lt_iff' (Nat.cast_pos.mpr (Nat.factorial_pos _))]
+
+theorem tendsto_pow_div_factorial (c : K) :
```

Comment:
This is a generalisation of `Real.tendsto_pow_div_factorial_atTop`, right? I'd say you should rename this to mention atTop, but also deprecate the Real version in favour of this one